appsec/src/extension/ddappsec.c

```diff
@@ -456,6 +456,49 @@ static PHP_FUNCTION(datadog_appsec_testing_request_exec)
     RETURN_TRUE;
 }
 
+static PHP_FUNCTION(datadog_appsec_push_address)
+{
+    UNUSED(return_value);
+    if (!DDAPPSEC_G(active)) {
+        mlog(dd_log_debug, "Trying to access to push_address "
+                           "function while appsec is disabled");
+        return;
+    }
+
+    zend_string *key = NULL;
+    zval *value = NULL;
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "Sz", &key, &value) == FAILURE) {
+        RETURN_FALSE;
+    }
+
+    zval parameters_zv;
+    zend_array *parameters_arr = zend_new_array(1);
+    ZVAL_ARR(&parameters_zv, parameters_arr);
+    zval *res = zend_hash_add(Z_ARRVAL(parameters_zv), key, value);
+    if (res == NULL) {
+        zval_ptr_dtor(&parameters_zv);
+        mlog_g(dd_log_debug, "Parameters could not be added");
+        return;
+    }
+    Z_TRY_ADDREF_P(value);
+
+    dd_conn *conn = dd_helper_mgr_cur_conn();
+    if (conn == NULL) {
+        zval_ptr_dtor(&parameters_zv);
+        mlog_g(dd_log_debug, "No connection; skipping push_address");
+        return;
+    }
+
+    dd_result result = dd_request_exec(conn, &parameters_zv);
+    zval_ptr_dtor(&parameters_zv);
+
+    if (result == dd_should_block) {
+        dd_request_abort_static_page();
+    } else if (result == dd_should_redirect) {
+        dd_request_abort_redirect();
+    }
```

Comment:
This is not correct for roadrunner. It should have the same logic as `dd_find_and_apply_verdict_for_user` (perhaps it would be a good idea to centralize it, though where it's not immediately clear).

Reply:
Thanks for pointing it out. Done.
Regarding the centralising this, I would do it in a separate PR